py/picca/data.py

```diff
@@ -186,42 +189,42 @@ def __init__(self,ll,fl,iv,thid,ra,dec,zqso,plate,mjd,fid,order, diff=None,reso=
         self.mean_z = (sp.power(10.,ll[len(ll)-1])+sp.power(10.,ll[0]))/2./lam_lya -1.0
 
 
-    def __add__(self,d):
+    def __add__(self, d):
 
-        if not hasattr(self,'ll') or not hasattr(d,'ll'):
+        if not hasattr(self, 'll') or not hasattr(d, 'll'):
             return self
 
-        ll = sp.append(self.ll,d.ll)
-        fl = sp.append(self.fl,d.fl)
-        iv = sp.append(self.iv,d.iv)
-        mmef = None
-        if self.mmef is not None:
-            mmef = sp.append(self.mmef,d.mmef)
+        dic = {}  # this should contain all quantities that are to be coadded with ivar weighting
 
-        bins = sp.floor((ll-forest.lmin)/forest.dll+0.5).astype(int)
-        cll = forest.lmin + sp.arange(bins.max()+1)*forest.dll
-        cfl = sp.zeros(bins.max()+1)
-        civ = sp.zeros(bins.max()+1)
-        if mmef is not None:
-            cmmef = sp.zeros(bins.max()+1)
-        ccfl = sp.bincount(bins,weights=iv*fl)
-        cciv = sp.bincount(bins,weights=iv)
-        if mmef is not None:
-            ccmmef = sp.bincount(bins,weights=iv*mmef)
-        cfl[:len(ccfl)] += ccfl
-        civ[:len(cciv)] += cciv
-        if mmef is not None:
-            cmmef[:len(ccmmef)] += ccmmef
-        w = (civ>0.)
+        ll = sp.append(self.ll, d.ll)
+        dic['fl'] = sp.append(self.fl, d.fl)
+        iv = sp.append(self.iv, d.iv)
 
+        if self.mmef is not None:
+            dic['mmef'] = sp.append(self.mmef, d.mmef)
+        if self.diff is not None:
+            dic['diff'] = sp.append(self.diff, d.diff)
+        if self.reso is not None:
+            dic['reso'] = sp.append(self.reso, d.reso)
+
+        bins = sp.floor((ll - forest.lmin) / forest.dll + 0.5).astype(int)
+        cll = forest.lmin + sp.arange(bins.max() + 1) * forest.dll
+        civ = sp.zeros(bins.max() + 1)
+        cciv = sp.bincount(bins, weights=iv)
+        civ[:len(cciv)] += cciv
```

Comment:
Let's not modify this line only for spaces

Reply:
I don't see where this comment belongs to anymore, I guess it's resolved now